src/hypothesis/internal/conjecture/minimizer.py

```diff
@@ -78,55 +86,95 @@ def _shrink_index(self, i, c):
             hbytes([255] * (self.size - i - 1))
         )
 
+    def ddlower(self, c):
+        self.ddfixate(lambda b: min(b, c))
+
+    def ddshift(self):
+        self.ddfixate(lambda b: b >> 1)
+
+    def ddsub(self):
+        self.ddfixate(lambda b: max(0, b - 1))
+
+    def ddfixate(self, f):
+        prev = -1
+        while prev != self.changes:
+            prev = self.changes
+            k = len(self.current)
+            while k > 0:
+                i = 0
+                while i + k <= len(self.current):
+                    self.incorporate(
+                        self.current[:i] +
+                        hbytes(f(b) for b in self.current[i:i + k]) +
+                        self.current[i + k:]
+                    )
+                    i += k
+                k //= 2
+
+    def rotate_suffixes(self):
+        for significant, c in enumerate(self.current):  # pragma: no branch
+            if c:
+                break
+        assert self.current[significant]
+
+        prefix = hbytes(significant)
+
+        for i in range(1, self.size - significant):
+            left = self.current[significant:significant + i]
+            right = self.current[significant + i:]
+            rotated = prefix + right + left
+            if rotated < self.current:
+                self.incorporate(rotated)
+
+    def shrink_indices(self):
+        for i in hrange(self.size):
+            t = self.current[i]
+            if t > 0:
+                ss = small_shrinks[self.current[i]]
+                for c in ss:
+                    if self._shrink_index(i, c):
+                        for c in hrange(self.current[i]):
```

Comment:
No, because if we hit this point then self.current has changed, so self.current[i] < t.